### ISSUE TITLE
Broken Autocomplete — Module Definitions: Polymorphic Type Aliases

### ISSUE THREAD

**Issue by mwalkerwells** [facebook/flow#4588]
**TL;DR:** If an export is an object and a value has a type which is polymorphic, type inference breaks ☹️
___
Took me a while to figure out why [my flow-typed library](https://github.com/flowtype/flow-typed/blob/master/definitions/npm/styled-components_v2.x.x/flow_v0.42.x-/styled-components_v2.x.x.js) was not providing autocomplete and type-a-head results & I finally narrowed it down to polymorphic type aliases...

It seems to apply to all library type export definitions (possibly more), flow-typed or not.
___
```js
// ./flowtype-definitions/
declare module 'foo-module' {
  declare type Bar<A> = { quux: 'quux' }
  declare type Baz = { quuux: 'quuux' }

  declare module.exports: {
    bar: Bar<void>,
    baz: Baz,
  }
}
```
```js
// ./foo.js
import foo from 'foo-module'

foo.ba // (See attached screenshots)
```
___
_So far so good, we see both results..._
  
<img width="338" alt="screen shot 2017-08-12 at 8 36 30 pm" src="https://user-images.githubusercontent.com/16157429/29245929-a54c18b2-7f9e-11e7-95dd-6b111da9e0f4.png">

_Huh...no autocomplete or type for bar?_
  
<img width="287" alt="screen shot 2017-08-12 at 8 36 48 pm" src="https://user-images.githubusercontent.com/16157429/29245928-a54b035a-7f9e-11e7-8cb4-538a2d073ab8.png">

_But ``baz`` works!_
  
<img width="411" alt="screen shot 2017-08-12 at 8 37 13 pm" src="https://user-images.githubusercontent.com/16157429/29245927-a5496c5c-7f9e-11e7-9139-c2eb18d21c39.png">

**Comment by mwalkerwells** [comment]
@jeffmo Given the implication for flow-typed, I'm curious if you know who to triage this to?

**Comment by amunra81** [comment]
It's about all generic types, not only those in module declarations

<img width="396" alt="image" src="https://user-images.githubusercontent.com/3176255/49500810-35604400-f87a-11e8-8dfc-70f060c07ae7.png">

 so far so good:

<img width="524" alt="image" src="https://user-images.githubusercontent.com/3176255/49500844-490baa80-f87a-11e8-8e3f-5c60dd50ff2e.png">

we see baz

<img width="577" alt="image" src="https://user-images.githubusercontent.com/3176255/49501002-9556ea80-f87a-11e8-83ee-211f078cda04.png">

but we can't see bar

<img width="213" alt="image" src="https://user-images.githubusercontent.com/3176255/49500984-8d974600-f87a-11e8-82bd-d08969ab224c.png">
